authenticators/cf_authenticator.go

```diff
@@ -83,7 +84,16 @@ func (cfa *CFAuthenticator) Authenticate(metadata ssh.ConnMetadata, password []b
 		return nil, err
 	}
 
-	processGuid, err := cfa.checkAccess(logger, appGuid, index, string(cred))
+	parts := strings.Split(cred, " ")
+	tokenString := parts[1]
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		return []byte("Doesntmatter"), nil
+	})
+
+	username := token.Claims["user_name"].(string)
+	principal := token.Claims["user_id"].(string)
+
+	processGuid, err := cfa.checkAccess(logger, appGuid, index, string(cred), username, principal)
```

Comment:
Instead of passing username and principal into checkAccess, it may be cleaner to insert this line before the call to checkAccess:

```
logger = logger.WithData(lager.Data{
                "app":       fmt.Sprintf("%s/%d", appGuid, index),
                "principal": principal,
                "username":  username,
 })
```

This would get rid of the code duplication, it would avoiding passing around extra arguments that are only used for logging, and it would also print out this information for other other failure cases during this step.